.github/workflows/python-release.yaml

```diff
@@ -0,0 +1,215 @@
+name: Python Release
+
+on:
+  push:
+    tags:
+      - 'v*'
+    branches-ignore:
+      - '*'
+  workflow_run:
+    workflows: [Python Test]
+    types:
+      - completed
+  workflow_dispatch:
+
+env:
+  # Please make sure this version is included in the `matrix`, as the
+  # `matrix` section can't use `env`, so it must be entered manually
+  DEFAULT_PYTHON_VERSION: '3.11'
+  # It would be nice to be able to also define a DEFAULT_UBUNTU_VERSION
+  # but sadly `env` can't be used either in `runs-on`.
+
+jobs:
+  publish-docs:
+    name: Publish documentation website to GitHub pages
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-20.04
+    permissions:
+      contents: write
```

Comment:
It would make the workflow a bit slower as they won't happen in parallel, but maybe it is worth publishing docs **only** if the build was successful? Otherwise we'll have public docs but no release.